bundles/org.openhab.binding.boschshc/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -606,4 +642,32 @@
 		</state>
 	</channel-type>
 
+	<channel-type id="key-code">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Key Code</label>
+		<description>Integer code of the key that was pressed.</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="key-name">
+		<item-type>String</item-type>
+		<label>Key Name</label>
+		<description>Name of the key that was pressed.</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="key-event-type">
+		<item-type>String</item-type>
+		<label>Key Event Type</label>
+		<description>Indicates how the key was pressed. Possible values are PRESS_SHORT, PRESS_LONG and PRESS_LONG_RELEASED.</description>
```

Comment:
For your consideration: possible states can be listed as an option:
https://www.openhab.org/docs/developer/bindings/thing-xml.html#state-description

Reply:
Thanks for the hint, I added options where applicable :+1: